### DISCUSSION TITLE
The version of vulnerability may have some problem.

### DISCUSSION THREAD

**Discussion by braxidas** [aquasecurity/trivy#4831] [Q&A]
### Description

"VulnerabilityID":"CVE-2022-42720"
"pkgID":"kernel-tools-libs@3.10.0-1160.92.1.e17.x86 64"
"pkaName"."kernel-tools 1ibs"
"Installedversion": "3,10.9-1160,92.1.e17"
"Layer":
"severitySource":"redhat"
"PrimaryURL":"https://avd.aquasec.com/nvd/cve-2022-42720"
"Title": "use-after-free in bss ref get in net/wireless/scan.c"
"decription": "arious refcounting bugs in the multi-BSS handling in the mac80211 stack in the Linx kemel 5.1 through 5.19x before 5.19.16 could
"Severity":"MEDIUM"

### Desired Behavior

The Vulnerability should not be detected because the its Installedversion is 3,10.9-1160,92.1.e17,  earlier than 5.1 as is decripted in its description. 

### Actual Behavior

I read your code and find any pkg whose Installedversion is less than fixed version will be detected, even its version is earlier than the version when the vulnerability appear. Maybe the database doesn‘t store relative data. Nearly 6k vulnerability's descrptions contain the start_verison of the vulnerability while there are 9w vulnerabilities. Can you tell me whether you know this problem and ignore it because fixing it will take a lot of efforts but bring a little improvement, or you haven't realized the problem, or you is still planning to fix it in the later version in an extremely efficient way. Thank you for your time.Your reply will be highly appercaited.

### Reproduction Steps

```bash
trivy fs . --scanners vuln --skip-db-update  -f json -o testsys.json --vuln-type os
in centos7.9
```

### Target

None

### Scanner

None

### Output Format

None

### Mode

None

### Debug Output

```bash
trivy fs . --scanners vuln --skip-db-update  -f json -o testsys.json --vuln-type os
```

### Operating System

centos7.9 

### Version

```bash
0.43.1
```

### Checklist

- [ ] Run `trivy image --reset`
- [ ] Read [the troubleshooting](https://aquasecurity.github.io/trivy/latest/docs/references/troubleshooting/)

**Comment by knqyf263** [comment]
Trivy follows security advisories from vendors. They have backports, so the older version could be affected. For example, they fixed CVE-2022-42720 in kernel-4.18.0-477.10.1.el8_8 even though it is not kernel 5.1. If you think it is wrong, you can contact Red Hat.
https://access.redhat.com/errata/RHSA-2023:2951

But, in your case, RHEL 7 is out of support scope. It means Red Hat doesn't know if it is affected. Trivy should not detect it for that reason. We'll fix it.

**Reply by braxidas** [reply]
Thank you very much.

**Reply by knqyf263** [reply]
After another read of the Red Hat documentation, we found that they say "out of support" products should be assumed to be affected.

>My product is listed as "Out of Support Scope". What does this mean?
>When a product is listed as "Out of Support Scope", it means a vulnerability with the impact level assigned to this CVE is no longer covered by its current [support lifecycle phase](https://access.redhat.com/product-life-cycles/update_policies). The product has been identified to contain the impacted component, but analysis to determine whether it is affected or not by this vulnerability was not performed. The product should be assumed to be affected. Customers are advised to apply any mitigation options documented on this page, consider removing or disabling the impacted component, or upgrade to a supported version of the product that has an update available.

For that reason, we will not change the behavior. You can ignore the vulnerability if you want.
https://aquasecurity.github.io/trivy/v0.43/docs/configuration/filtering/
